File: src/FMS.Site/Models/Match.cs
﻿using FMS.Site.Data;

namespace FMS.Site.Models
{
    public class Match
    {
        public int Id { get; set; }
        public int SeasonId { get; set; }
        public int WeekId { get; set; }
        public int DivisionId { get; set; }
        public string Completed { get; set; }

        public int HomeTeamId { get; set; }
        public int AwayTeamId { get; set; }
        public int HomeTeamScore { get; set; }
        public int AwayTeamScore { get; set; }

        public string HomeTeam => TeamData.GetTeamById(HomeTeamId).Name;
        public string AwayTeam => TeamData.GetTeamById(AwayTeamId).Name;

        public int CurrentWeek => GameData.CurrentWeek;
        public string Season => SeasonData.GetById(SeasonId).Name;

        // public List<MatchEvent> matchEvents { get; set; }
    }
}

File: src/FMS.Site/src/components/matches/presentation/match.js
import React, { PropTypes } from "react";

import Row from "../../table/presentation/row";
import Data from "../../table/presentation/data";
import Button from "../../button/presentation";

const Match = ({ match, playMatch }) => {

    return (
        <Row>
            <Data>{match.weekId}</Data>
            {match.completed == "No" ?
            <Data>{match.homeTeam + " : " + match.awayTeam}</Data>
            :
            <Data>{match.homeTeam + " " + match.homeTeamScore + " : " + match.awayTeamScore + " : " + match.awayTeam}</Data>
            }

            <Data>{match.completed}</Data>
            { match.weekId <= match.currentWeek ?
                <Data>
                {match.completed == "No" ? 
                    <Button onClick={playMatch}>Play Match</Button>
                : <Button state="success" onClick={playMatch}>View Match</Button> 
                }</Data>
            : null }
        </Row>
    );
};

Match.propTypes = {
    match: PropTypes.object.isRequired,
    playMatch: PropTypes.func.isRequired
};

export default Match;